controllers/config_map.go

```diff
@@ -224,31 +310,48 @@ func (r *ConfigMapReconciler) reconcileDelete(ctx context.Context, cluster *capa
 	return ctrl.Result{}, nil
 }
 
-func (r *ConfigMapReconciler) AddFinalizer(ctx context.Context, awsCluster *capa.AWSCluster) error {
-	originalCluster := awsCluster.DeepCopy()
-	controllerutil.AddFinalizer(awsCluster, Finalizer)
-	return r.Client.Patch(ctx, awsCluster, client.MergeFrom(originalCluster))
+func (r *ConfigMapReconciler) AddFinalizer(ctx context.Context, capiCluster *capi.Cluster) error {
+	originalCluster := capiCluster.DeepCopy()
+	controllerutil.AddFinalizer(capiCluster, Finalizer)
+	return r.Client.Patch(ctx, capiCluster, client.MergeFrom(originalCluster))
 }
 
-func (r *ConfigMapReconciler) RemoveFinalizer(ctx context.Context, awsCluster *capa.AWSCluster) error {
-	originalCluster := awsCluster.DeepCopy()
-	controllerutil.RemoveFinalizer(awsCluster, Finalizer)
-	return r.Client.Patch(ctx, awsCluster, client.MergeFrom(originalCluster))
+func (r *ConfigMapReconciler) RemoveFinalizer(ctx context.Context, capiCluster *capi.Cluster) error {
+	originalCluster := capiCluster.DeepCopy()
+	controllerutil.RemoveFinalizer(capiCluster, Finalizer)
+	error := r.Client.Patch(ctx, capiCluster, client.MergeFrom(originalCluster))
```

Comment:
There should only be `err`, not `error`. The `Cluster` finalizer removal+patch must come _last_ in this function, since if something goes wrong afterwards with the `AWSCluster` logic, we're not watching the `Cluster` anymore, so no further calls may happen.